comp/updater/telemetry/component.go

```diff
@@ -0,0 +1,16 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// Package telemetry provides the installer telemetry component.
+package telemetry
+
+import "github.com/DataDog/datadog-agent/pkg/installer"
+
+// team: fleet
+
+// Component is the component type.
+type Component interface {
+	installer.Telemetry
```

Comment:
`Start` and `Stop` methods should not be part of a public interface. We don't want multiple sources handling the lifecycle.

And in general we want `Component` interfaces to document all their methods. Embedding an interface in another does go against that design. We want users of a component to easily know what a component offers and how to use it.

For this case in particular, do we need the exposed method from `installer.Telemetry` (I don't see them used anywhere) ? Since they are added to the Fx lifecycle do we need access to them from the outside ?

Reply:
Removed the Start/Stop. It's fine in this context since telemetry only needs those methods but this pattern of embedding interface is the only way we have found to avoid tying ourselves strongly with the component framework. We need to avoid this as we'll have to import part of the installer code in the operator which does not use fx.

Reply:
We have an [RFC](https://docs.google.com/document/d/1sPXUviG-RvMCUvqdwfUYbTOTV05Z_y9MuwlYcyOdG0I/edit#heading=h.o5gstqo08gu5) to have a defacto way to work with components and not having to import FX as part of it if we want to extract such functionality into a separate project. 

Outside of that I believe the RC and telemetry component would be a good candidate for using that pattern. Could you explain with a bit more details why adding Fx to operator would be an issue? 

cc @dustmop @hush-hush @ogaca-dd

Reply:
Until at least DASH we're on very tight timelines with this project so it's hard for us to take on extra work of any kind if it can be avoided. After we're in a more stable state and that RFC is approved we can try that new layout

Reply:
The remote config service is probably the right target tho if you want to put some time on the ASC side to try your RFC on something concrete. It's imported in the operator and our backend + has its own go.mod